### PR TITLE
feat: added tree traversal for ranker

### PR DESCRIPTION
**Pull request by maximilianwerk**
Adaptions for the rankers.

### CODE REVIEW

**Comment by JoanFM** [comment]
Is the integration test on level depth passing without skip?

File: jina/drivers/rank.py [hunk 1]
```diff
@@ -59,10 +57,6 @@ def _apply_all(self, docs: Iterable['jina_pb2.Document'], context_doc: 'jina_pb2
         :return:
         """
 
-        # if at the top-level already, no need to aggregate further
-        if context_doc is None:
```
Comment (JoanFM) [inline jina/drivers/rank.py:63]: lots of love here!!!!

**Comment by maximilianwerk** [comment]
No, that will only work, when some more drivers are touched. ;) Be patient, I promise, that it will be runable, once we are done with all drivers.

File: jina/drivers/rank.py [hunk 2]
```diff
@@ -130,7 +124,7 @@ def _apply_all(self, docs: Iterable['jina_pb2.Document'], context_doc: 'jina_pb2
         query_chunk_meta = {}
         match_chunk_meta = {}
         # doc_id_to_match_map = {}
-        for match in context_doc.matches:
```
Comment (JoanFM) [inline jina/drivers/rank.py:133]: lots of love here

File: jina/drivers/rank.py [hunk 3]
```diff
@@ -163,14 +157,15 @@ class Matches2DocRankDriver(BaseRankDriver):
...
-        super().__init__(recur_on='matches', *args, **kwargs)
+    def __init__(self, reverse: bool = False, traversal_paths: Iterable[str] = ['m'], *args, **kwargs):
+        super().__init__(use_tree_traversal=True, traversal_paths=traversal_paths, *args, **kwargs)
         self.reverse = reverse
 
     def _apply_all(self, docs: Iterable['jina_pb2.Document'], context_doc: 'jina_pb2.Document', *args, **kwargs) -> None:
         """ Call executer for score and sort afterwards here. """
 
         # if at the top-level already, no need to aggregate further
+        print(len(docs))
```
Comment (JoanFM) [inline jina/drivers/rank.py:168]: remember to remove

**Comment by JoanFM** [comment]
> No, that will only work, when some more drivers are touched. ;) Be patient, I promise, that it will be runable, once we are done with all drivers.

Ok!!!!

**Review by JoanFM — APPROVED** [review]

**Pull request merged**